OSDOCS-10553: Adds release notes for both perspectives of web console (#76907)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

[id="ocp-4-16-web-console_{context}"]
=== Web console

[id="ocp-4-16-patternfly4-deprecated"]
==== Patternfly 4 is now deprecated with 4.16
With this release, Patternfly 4 and React Router 5 is deprecated in the web console. All plugins should migrate to Patternfly 5 and React Router 6 as soon as possible.

[id="ocp-4-16-administrator-perspective_{context}"]
==== Administrator perspective

This release introduces the following updates to the *Administrator* perspective of the web console:

> * A Google Cloud Platform (GCP) token authorization was added to the *Infrastructure features* filter in the OperatorHub.

Reviewer: Shall we be more explicit here? The filter text is `Auth Token GCP` 
![Screenshot 2024-06-13 at 9 37 00 AM](https://github.com/openshift/openshift-docs/assets/12692381/d9bebf8c-a335-4013-b238-f99aabd1f944)

Reviewer: we also have a new `Configurable TLS ciphers` filter added from 4.16, the PR is https://github.com/openshift/console/pull/13555 
Shall we highlight in release note?

> * A new quick start is available with information on the impersonate workflow and functionality for users with impersonare access.

Reviewer: also shall we put the new quick start name here? It guides user how to impersonate `system:admin` user 
![Screenshot 2024-06-13 at 9 39 15 AM](https://github.com/openshift/openshift-docs/assets/12692381/1aca34a7-4a1e-4380-95bd-cc83cd4e246b)

[id="ocp-4-16-web-console"]
=== Web console

[id="ocp-4-16-patternfly4-deprecated"]
==== Patternfly 4 is now deprecated with 4.16

> With this release, Patternfly 4 and React Router 5 is depreciated in the web console. All plugins should migrate to Patternfly 5 and React Router 6 as soon as possible.

Reviewer 2: ```suggestion
With this release, Patternfly 4 and React Router 5 is deprecated in the web console. All plugins should migrate to Patternfly 5 and React Router 6 as soon as possible.
```

[id="ocp-4-16-administrator-perspective"]
==== Administrator perspective

This release introduces the following updates to the *Administrator* perspective of the web console:

* A Google Cloud Platform (GCP) token authorization was added to the *Infrastructure features* filter in the OperatorHub.
* A new quick start is available with information on the impersonate workflow and functionality for users with impersonare access.

[id="ocp-4-16-developer-perspective"]
==== Developer Perspective

This release introduces the following updates to the *Developer* perspective of the web console:

* When searching,

> A new section was added to the *Search* menu to display the recently searched items in the order they were searched.

Reviewer 3: Lgtm from my side :)

> A new section was added to the *Search* menu to display the recently searched items in the order they were searched.

Reviewer 4: Can we be more specific about the search menu?
```suggestion
A new section was added to the *Resources* dropdown on the search page to display the recently searched items in the order they were searched.
```

Author: Yes I like that! Thanks

* With this release, you can collapse and expand the *Getting started* section.

[id="ocp-4-16-console-telemetry-plugin"]
===== Console Telemetry plugin

With this release, you can configure your settings to opt in or out to sending Telemetry data to Red Hat. No data will be sent without choosing to opt in.

> 

Reviewer 5: @jerolimov Do we need to add this in release note since we disabled this feature using flag?

Reviewer 6: @lokanandaprabhu @opayne1 Yes, we should adjust this comment.

The UI is only visible if a customer adds a particular configuration to the cluster. That's not the default. Users will not see a UI for opt in until the cluster admins configure that option.

I think we should mention telemetry in the release note. But we should mention here that we enabled telemetry now also in the front end (= analytics) when the cluster telemetry is enabled.

```suggestion
===== Console Telemetry

With this release, we enabled anonymized frontend telemetry (user analytics) if cluster telemetry is enabled. This is the default for most of the cluster and helps our teams understand how the Console is used.

Cluster administrators can change this per cluster. They can switch to an opt-in, opt-out, or disable this frontend telemetry.
```

Wdyt @lokanandaprabhu @opayne1 @alimobrem @jhadvig ?

/cc @vikram-raj @invincibleJai

Author: Updated. Thanks everyone!

[id="ocp-4-16-web-console_{context}"]
=== Web console

[id="ocp-4-16-web-console-language"]
==== Language support for French and Spansih
With this release, French and Spanish are suported in the web console. You can update the language in the webconsole from the *Language* list on the *User Preferences* page.

[id="ocp-4-16-patternfly4-deprecated"]
==== Patternfly 4 is now deprecated with 4.16
With this release, Patternfly 4 and React Router 5 is deprecated in the web console. All plugins should migrate to Patternfly 5 and React Router 6 as soon as possible.

[id="ocp-4-16-administrator-perspective_{context}"]
==== Administrator perspective

This release introduces the following updates to the *Administrator* perspective of the web console:

* A Google Cloud Platform (GCP) token authorization `Auth Token GCP`, and a `Configurable TLS ciphers` filter was added to the *Infrastructure features* filter in the OperatorHub.

> * A new quick start, *Build with guided documentation*, is available with information on impersonating the `system:admin` user.

Reviewer: <img width="375" alt="Screenshot 2024-06-18 at 11 19 25 AM" src="https://github.com/openshift/openshift-docs/assets/12692381/70d53039-937b-4b57-a26c-6676e3ad3187">
The new quick start name is `Impersonating the system:admin user`

* A pod's last termination state is now available to view on the *Container list* and *Container details* pages.
* An `Impersonate Group` action is now available from the *Groups* and *Group details* pages without having to search for the appropriate `RoleBinding`.

[id="ocp-4-16-node-csr-handling"]
===== Node CSR handling in the {product-title} web console
With this release, the {product-title} web console supports node certificate signing requests (CSRs).

[id="ocp-4-16-cross-storage-class-clone-restore"]
===== Cross Storage Class clone and restore

> With this release, you can choose a storage class from the same provider when cloning or restoring. You can move from storage class with a replica of 3 to a replica of 2/1.

Reviewer: maybe we should add a word `different` then it becomes `you can choose a different storage class from the same ........`

> With this release, you can choose a storage class from the same provider when cloning or restoring. You can move from storage class with a replica of 3 to a replica of 2/1.

Reviewer: maybe we can use the text from PR? 
```
Users can now choose a storage class from the same provider when cloning or restoring.
This flexibility allows seamless transitions between storage classes with different replica counts (e.g., moving from SC with 3 replicas to 2/1 replicas).
```

Author: Updated

[id="ocp-4-16-developer-perspective_{context}"]
==== Developer Perspective

This release introduces the following updates to the *Developer* perspective of the web console:

* When searching, a new section was added to the list of *Resources* on the *Search* page to display the recently searched items in the order they were searched.
* With this release, you can collapse and expand the *Getting started* section.

[id="ocp-4-16-console-telemetry"]
===== Console Telemetry

> With this release, we enabled user analytics with anonymized frontend telemetry if cluster telemetry is enabled. This is the default for most of the cluster and provides Red Hat with metrics for how the Console is used. Cluster administrators can update this in each cluster and opt-in, opt-out, or disable frontend telemetry.

Reviewer 6: This sentence is correct.

I just thought that  "we enabled user analytics with anonymized frontend telemetry"  sounds "complicated".

Wdyt about this suggestion?

```suggestion
With this release, we enabled anonymized user analytics if cluster telemetry is enabled. This is the default for most of the cluster and provides Red Hat with metrics for how the Console is used. Cluster administrators can update this in each cluster and opt-in, opt-out, or disable frontend telemetry.
```

If you prefer your version that's also fine. :smirk:

Author: I like the suggestion! I was only trying to avoid using parenthesis since we only use them in specific cases.

Author: Updated

[id="ocp-4-16-web-console_{context}"]
=== Web console

[id="ocp-4-16-web-console-language"]
==== Language support for French and Spansih

> With this release, French and Spanish are suported in the web console. You can update the language in the webconsole from the *Language* list on the *User Preferences* page.

Reviewer 7: ```suggestion
With this release, French and Spanish are suported in the web console. You can update the language in the web console from the *Language* list on the *User Preferences* page.
```

[id="ocp-4-16-patternfly4-deprecated"]

> ==== Patternfly 4 is now deprecated with 4.16

Reviewer 7: ```suggestion
==== Patternfly 4 is now deprecated
```

Author: Keeping the version but changing to {product-version} want to avoid  any confusion about what versions are available where.

> With this release, Patternfly 4 and React Router 5 is deprecated in the web console. All plugins should migrate to Patternfly 5 and React Router 6 as soon as possible.

Reviewer 7: ```suggestion
With this release, Patternfly 4 and React Router 5 are deprecated in the web console. All plugins should migrate to Patternfly 5 and React Router 6 as soon as possible.
```

[id="ocp-4-16-administrator-perspective_{context}"]
==== Administrator perspective

This release introduces the following updates to the *Administrator* perspective of the web console:

> * A Google Cloud Platform (GCP) token authorization `Auth Token GCP`, and a `Configurable TLS ciphers` filter was added to the *Infrastructure features* filter in the OperatorHub.

Reviewer 7: ```suggestion
* A {gcp-first} token authorization, `Auth Token GCP`, and a `Configurable TLS ciphers` filter was added to the *Infrastructure features* filter in the OperatorHub.
```

* A new quick start, *Impersonating the system:admin user*, is available with information on impersonating the `system:admin` user.
* A pod's last termination state is now available to view on the *Container list* and *Container details* pages.
* An `Impersonate Group` action is now available from the *Groups* and *Group details* pages without having to search for the appropriate `RoleBinding`.

[id="ocp-4-16-node-csr-handling"]
===== Node CSR handling in the {product-title} web console
With this release, the {product-title} web console supports node certificate signing requests (CSRs).

[id="ocp-4-16-cross-storage-class-clone-restore"]
===== Cross Storage Class clone and restore

> With this release, you can choose a storage class from the same provider when cloning or restoring. This flexibility allows seamless transitions between storage classes with different replica counts. For example, moving from a storage class with 3 replicas to 2/1 replicas.

Reviewer 7: ```suggestion
With this release, you can choose a storage class from the same provider when completing clone or restore operations. This flexibility allows seamless transitions between storage classes with different replica counts. For example, moving from a storage class with 3 replicas to 2/1 replicas.
```

[id="ocp-4-16-developer-perspective_{context}"]
==== Developer Perspective

This release introduces the following updates to the *Developer* perspective of the web console:

* When searching, a new section was added to the list of *Resources* on the *Search* page to display the recently searched items in the order they were searched.
* With this release, you can collapse and expand the *Getting started* section.

[id="ocp-4-16-console-telemetry"]
===== Console Telemetry

> With this release, we enabled anonymized user analytics if cluster telemetry is enabled. This is the default for most of the cluster and provides Red Hat with metrics for how the Console is used. Cluster administrators can update this in each cluster and opt-in, opt-out, or disable frontend telemetry.

Reviewer 7: I would not usually see "we" so maybe Red{nbsp}Hat should be stated instead?

Author: I rephrased. Thanks!

> With this release, we enabled anonymized user analytics if cluster telemetry is enabled. This is the default for most of the cluster and provides Red Hat with metrics for how the Console is used. Cluster administrators can update this in each cluster and opt-in, opt-out, or disable frontend telemetry.

Reviewer 7: ```suggestion
With this release, we enabled anonymized user analytics if cluster telemetry is enabled. This is the default for most of the cluster and provides Red Hat with metrics for how the web console is used. Cluster administrators can update this in each cluster and opt-in, opt-out, or disable frontend telemetry.
```
